### ISSUE TITLE
[BUG] TorchForecastingModel.load() fails to load old models

### ISSUE THREAD

**Issue by alexcolpitts96** [unit8co/darts#1424]
This bug doesn't need to be fixed; I am mostly making its existance known. Commit cdf2b44804bbb35eaf9d43592e319d9a77958b6c has broken model loading with previous versions.

It might be beneficial to add a check when loading models to try and handle the previous format. 

Details:
Python 3.10.8
Darts 0.22.0@d6a74c00128f5897fe0bb068e11b7c4540799d66 -> cdf2b44804bbb35eaf9d43592e319d9a77958b6c

**Comment by dennisbader** [comment]
Hi @alexcolpitts96 and thanks for writing.

We are aware that the changes to model saving/loading will break loading TorchForecastingModels from previous versions.

It ultimately comes from pickling issues when saving loading across different versions (python/libraries/source code).

This is a sacrifice we take for the improved saving/loading behavior in the future versions.

In darts<=0.22.0, apart from saving only the state dict/checkpoint of the underlying torch model, we pickled the entire TorchForecastingModel including the PyTorch Lightning Trainer and the Model (module).

In darts>=0.23.0, we only pickle the TFM without Trainer and Model -> breaking. Plus we added some extra features like mapping to a specific device while loading.

To avoid breaking older saves when adapting the in the future, we could avoid pickling entirely and instead write for example a `config.json` file with everything the TorchForecastingModel needs to load a model correctly.

**Comment by alexcolpitts96** [comment]
I have spent a little more time figuring exactly what is causing the break and it wasn't the save/load update. 

I made a workaround in my downstream tooling to handle model different model version loading, but ultimately it came back to a change in pytorch-lightning. Models created with pytorch-lightning==1.8.3 wouldn't load for 1.8.4+.

I haven't had time to dig through the change logs but it is not a darts issues.
